### DISCUSSION TITLE
Use Absolute paths and filepaths everywhere

### DISCUSSION THREAD

**Discussion by Arthapz** [xmake-io/xmake#1935] [Ideas]
It's easier to work with absolute path in IDE plugin side, and it would be more consistent cause xmake source batches are absolute filepaths

**Comment by waruqi** [comment]
It will cause many other problems, such as easy to trigger win path length limitation, the generated vsproj and makefile files cannot be moved, and the absolute path with spaces can easily trigger some unknown problems.

you can pass work directory to run xmake

**Reply by waruqi** [reply]
You can refer to the xmake-vscode plugin, I have never encountered path problems
